ReactiveFeedback/Feedback.swift

```diff
@@ -5,86 +5,108 @@ import enum Result.NoError
-    /// and is different from the previous one, otherwise, it cancels any previously performed effects
+    /// Creates a Feedback which re-evaluates the given effect every time the
+    /// `Signal` derived from the latest state yields a new value.
+    ///
+    /// If the previous effect is still alive when a new one is about to start,
+    /// the previous one would automatically be cancelled.
     ///
     /// - parameters:
-    ///     - query: A closure which defines for which value perform the effect
-    ///     - effects: A sequence of the Events over time that mutate the State
-    public init<Control: Equatable, Effect: SignalProducerConvertible>(
-        query: @escaping (State) -> Control?,
-        effects: @escaping (Control) -> Effect
+    ///   - transform: The transform which derives a `Signal` of values from the
+    ///                latest state.
+    ///   - effects: The side effect accepting transformed values produced by
+    ///              `transform` and yielding events that eventually affect
+    ///              the state.
+    public init<U, Effect: SignalProducerConvertible>(
+        deriving transform: @escaping (Signal<State, NoError>) -> Signal<U, NoError>,
+        effects: @escaping (U) -> Effect
     ) where Effect.Value == Event, Effect.Error == NoError {
         self.events = { scheduler, state in
-            return state
-                .map(query)
-                .skipRepeats { $0 == $1 }
-                .flatMap(.latest) { control -> SignalProducer<Event, NoError> in
-                    guard let control = control else { return .empty }
-                    return effects(control).producer
-                        .observe(on: scheduler)
-                }
+            return transform(state)
+                .flatMap(.latest, effects)
+                .observe(on: scheduler)
         }
     }
 
-    /// Creates a Feedback which will perform effects when `query` exists (i.e not nil)
-    /// otherwise, it cancels any previously performed effects
+    /// Creates a Feedback which re-evaluates the given effect every time the
+    /// transform yields a non-`nil` distinct value from the last yielded value.
+    ///
+    /// If the previous effect is still alive when a new one is about to start,
+    /// the previous one would automatically be cancelled.
+    ///
+    /// - parameters:
+    ///   - transform: The transform to apply on the state.
+    ///   - effects: The side effect accepting transformed values produced by
+    ///              `transform` and yielding events that eventually affect
+    ///              the state.
+    public init<Control: Equatable, Effect: SignalProducerConvertible>(
+        skippingRepeated transform: @escaping (State) -> Control?,
+        effects: @escaping (Control) -> Effect
+    ) where Effect.Value == Event, Effect.Error == NoError {
+        self.init(deriving: { $0.filterMap(transform).skipRepeats(==) },
```

Comment:
This is not the same as `query`. Because previous implementation cancels previous `Effect` when `Control` switches to `nil`

Reply:
This is actual reason why it's called `Control` because value of the `State` acts as a `Feedback` controller :)